Updated the EOS functions to scipy format

nhew1994 opened the pull request:
I tested it and double checked the equations. Should be correct, but doesn't hurt to check again. Works for me, If you can send me a few of your configs for me to test as well that would be great.

nhew1994 committed:
Updated the EOS functions to scipy format

nhew1994 committed:
Merge branch 'main' into dev/nigelhew

nhew1994 committed:
Update dfttk/eos_fit.py
Delete an extra comma

nhew1994 committed:
Update dfttk/eos_fit.py
Cleaned up the EOS functions even further

lukeamyers reviewed (approved):
I tried it out, with one of my configs. It works, but something has changed with fit_to_all_eos() when running plot_ev() that gives the below warning. 
```
/storage/work/lam7027/bin/vasp-job-automation/dfttk/eos_fit.py:631: FutureWarning:

The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.

```
seems there is empty data.

This function is not used in an ideal way anyhow. It doesn't make sense to fit all of them if we only care about one eos. plot_ev() could be updated to work more directly if you feel like it.

nhew1994 committed:
Update dfttk/eos_fit.py
Changed fit_to_all_eos to append a list of dataframes first, then concat them later. This
gets rid of the deprecation warning.
